runtime/backtrace.c

```diff
@@ -358,7 +358,7 @@ CAMLprim value caml_get_exception_backtrace(value unit)
       Store_field(arr, i, caml_convert_debuginfo(dbg));
     }
 
-    res = caml_alloc_small(1, 0); Field(res, 0) = arr; /* Some */
+    res = caml_alloc_small(1, Tag_some); Field(res, 0) = arr;
```

Comment:
I wondered if we should consider using `Some_val(res) = arr` instead of `Field(res, 0) = arr`, but I guess that it's best to keep the macro for reading only, not for writing.

Reply:
I like this idea, it would be useful as a example of the macro being usable in lhs as well as rhs of statements (but I'm not doing it unless I get the green light ;-) )

4.12 also introduced [`caml_alloc_some`](https://github.com/ocaml/ocaml/blob/trunk/runtime/alloc.c#L271-L277) to allocate a `Some x` block, that I haven't used here.

Reply:
> I wondered if we should consider using Some_val(res) = arr instead of Field(res, 0) = arr

I much prefer the use of `caml_alloc_some`.  Macros that look like functions but happen to be l-values make me nervous.  (What if one day they become functions and stop being l-values?)

Reply:
Pushed a new revision where I used `caml_alloc_some` in the 3 places options were allocated.